tst/EnergyPlus/unit/WindowAC.unit.cc

```diff
@@ -514,3 +514,487 @@ TEST_F(EnergyPlusFixture, WindowAC_VStest1)
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 6;    // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesPerTimeStep = 10;    // must initialize this to get schedules initialized
+    ScheduleManager::ProcessScheduleInput(*state); // read schedule data
+
+    bool errorsFound(false);
+    HeatBalanceManager::GetProjectControlData(*state, errorsFound); // read project control data
+    EXPECT_FALSE(errorsFound);
+    // OutputProcessor::TimeValue.allocate(2);
+    state->dataGlobal->DDOnlySimulation = true;
+
+    SimulationManager::GetProjectData(*state);
+    OutputReportPredefined::SetPredefinedTables(*state);
+    HeatBalanceManager::SetPreConstructionInputParameters(*state); // establish array bounds for constructions early
+
+    state->dataGlobal->BeginSimFlag = true;
+    state->dataGlobal->BeginEnvrnFlag = true;
+    state->dataGlobal->ZoneSizingCalc = true;
+    EnergyPlus::createFacilityElectricPowerServiceObject(*state);
+
+    SizingManager::ManageSizing(*state);
+
+    SimulationManager::SetupSimulation(*state, errorsFound);
+    //
+
+    Real64 qDotMet(0.0);    // Watts total cap
+    Real64 lDotProvid(0.0); // latent removal kg/s
+    int compIndex(0);
+    WindowAC::SimWindowAC(*state, "ZONE1WINDAC", 1, true, qDotMet, lDotProvid, compIndex);
+    // check input processing
+    EXPECT_EQ(compIndex, 1);
+
+    double constexpr expected_full_airflow = 0.041484382187390034;
+    double constexpr scaler_cooling_saf = 0.5;
+    double constexpr scaler_cooling_cap = 0.5;
+    double constexpr expected_full_cap = 622.50474573886743;
+
+    auto const &windowAC = state->dataWindowAC->WindAC(compIndex);
+
+    ASSERT_GT(windowAC.FanIndex, 0);
+    auto const &fan = state->dataFans->fans(windowAC.FanIndex);
+
+    auto const &finalZoneSizing = state->dataSize->FinalZoneSizing(1);
+
+    EXPECT_NEAR(expected_full_airflow, finalZoneSizing.DesCoolVolFlow, 0.0001);
+
+    ASSERT_GT(windowAC.HVACSizingIndex, 0);
+    auto const &zoneHVACSizing = state->dataSize->ZoneHVACSizing(windowAC.HVACSizingIndex);
+
+    EXPECT_EQ(DataSizing::FractionOfAutosizedCoolingAirflow, zoneHVACSizing.CoolingSAFMethod);
+    EXPECT_EQ(scaler_cooling_saf, zoneHVACSizing.MaxCoolAirVolFlow);
+    EXPECT_EQ(DataSizing::FractionOfAutosizedCoolingCapacity, zoneHVACSizing.CoolingCapMethod);
+    EXPECT_EQ(scaler_cooling_cap, zoneHVACSizing.ScaledCoolingCapacity);
+
+    EXPECT_EQ(windowAC.DXCoilType_Num, HVAC::Coil_CoolingAirToAirVariableSpeed);
+    ASSERT_GT(windowAC.DXCoilIndex, 0);
+    auto const &varSpeedCoil = state->dataVariableSpeedCoils->VarSpeedCoil(windowAC.DXCoilIndex);
+
+    // check Sizing
+    EXPECT_NEAR(expected_full_airflow * scaler_cooling_saf, windowAC.MaxAirVolFlow, 0.0001);
+    EXPECT_NEAR(expected_full_airflow * scaler_cooling_saf, fan->maxAirFlowRate, 0.0001);
+
+    EXPECT_NEAR(622.505, varSpeedCoil.RatedCapCoolTotal, 0.001);
```

Comment:
Note that the Coil:Cooling:DX:VariableSpeed is not affected.

* I don't know if we would want that.
* SizeVarSpeedCoil seem to not call the Autosizing modules. Instead, it uses the FinalZoneSizing's DesCoolVolFlow / DesCoolHeatFlow. cf https://github.com/NREL/EnergyPlus/blob/f274ccbcef211face33328309188b29b36fd0668/src/EnergyPlus/VariableSpeedCoils.cc#L4715-L4717

Reply:
Filed at https://github.com/NREL/EnergyPlus/issues/10699